sklearn/preprocessing/data.py

```diff
@@ -2369,16 +2389,47 @@ def _transform_col(self, X_col, quantiles, inverse):
 
         return X_col
 
+    def _assert_finite_or_nan(self, X):
+        """Check that X contain finite or NaN values."""
+        X = np.asanyarray(X)
+        # First try an O(n) time, O(1) space solution for the common case that
+        # everything is finite; fall back to O(n) space np.isfinite to prevent
+        # false positives from overflow in sum method.
+        if (X.dtype.char in np.typecodes['AllFloat']
+                and not np.isfinite(X[~np.isnan(X)].sum())
+                and not np.isfinite(X[~np.isnan(X)]).all()):
+            raise ValueError("Input contains infinity"
+                             " or a value too large for %r." % X.dtype)
+        if np.count_nonzero(self._mask_missing_values):
+            if LooseVersion(np.__version__) >= '1.9':
+                self._percentile_func = np.nanpercentile
+            else:
+                raise NotImplementedError(
+                    'QuantileTransformer does not handle NaN value with'
```

Comment:
Is it easy enough to just implement in sklearn.utils.fixes:

```
def nanpercentile(a, q):
    return np.percentile(np.compress(a, ~np.isnan(a)), q)
```

seeing as we don't use the other features of nanpercentile?

It is annoying to have parts of the library with different minimum numpy requirements. It means that code is not portable across supported platforms.

Reply:
Right